src/main/java/bio/terra/service/filedata/azure/util/BlobContainerClientFactory.java

```diff
@@ -0,0 +1,201 @@
+package bio.terra.service.filedata.azure.util;
+
+import com.azure.core.credential.AzureSasCredential;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.storage.blob.*;
+import com.azure.storage.blob.models.UserDelegationKey;
+import com.azure.storage.blob.sas.BlobSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import com.azure.storage.common.sas.SasProtocol;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Locale;
+import java.util.Objects;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * A class that wraps the SDK container client and facilities the creation of SAS Urls for blobs by
+ * abstracting the different strategies that depend on the authentication mechanism.
+ */
+public class BlobContainerClientFactory {
+
+  public static final int EXPIRATION_DAYS = 7;
+  private static final HttpClient httpClient = HttpClient.createDefault();
```

Comment:
In general, `static final` fields are constants and should be written in uppercase. Also note that this requires that `HttpClient` is thread-safe. I looked at the source code and the javadoc and I didn't any indication about this one way or the other.
```suggestion
  private static final HttpClient HTTP_CLIENT = HttpClient.createDefault();
```

Reply:
HttpClient is thread safe, and must be shared to benefit from the underlying connection pooling. Further thinking about this, there is little chance that a source and destination will benefit from using the same connection pool. So I will make the field just `final`. This will make the naming suggestion moot.